powerlevel10K_rainbow has spurious space at the end of the second line.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

At some point in the last couple versions the flush right block on the second line of powerlevel10K_rainbow has an extraneous space at the end, preventing the the glyphs from lining up.

<img width="706" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/8260876/08ca8077-7675-4c27-abec-573baac7887b">


### Theme

powerlevel10K_rainbow

### What OS are you seeing the problem on?

macOS

### Which shell are you using?

zsh

### Log output

```Shell
Segments:

ConsoleTitle(true)                         -   0 ms
Os(true)                                   -   0 ms
Path(true)                                 -   0 ms
Git(false)                                 -   0 ms
Node(false)                                -   0 ms
Go(false)                                  -   0 ms
Julia(false)                               -   0 ms
Python(false)                              -   0 ms
Ruby(false)                                -   0 ms
Azfunc(false)                              -   0 ms
Aws(false)                                 -   0 ms
Root(false)                                -   0 ms
Executiontime(false)                       -   0 ms
Status(true)                               -   0 ms
Time(true)                                 -   0 ms
Text(true)                                 -   0 ms
Text(true)                                 -   0 ms

Run duration: 45.193792ms

Cache path: /Users/martin.diers/.cache/oh-my-posh

Config path: /opt/homebrew/opt/oh-my-posh/themes/powerlevel10k_rainbow.omp.json

Logs:

[DEBUG] 13:32:47.154 shell.go:Getenv:392 → iTerm.app
[TRACE] 13:32:47.154 shell.go:Getenv(TERM_PROGRAM) - 101.875µs
[DEBUG] 13:32:47.155 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.155 shell.go:Getenv(XDG_CACHE_HOME) - 2.792µs
[TRACE] 13:32:47.155 shell_unix.go:CachePath() - 20µs
[DEBUG] 13:32:47.155 shell.go:Getenv:392 → /opt/homebrew/opt/oh-my-posh/themes/powerlevel10k_rainbow.omp.json
[TRACE] 13:32:47.155 shell.go:Getenv(POSH_THEME) - 5.708µs
[DEBUG] 13:32:47.155 shell_unix.go:Platform:77 → darwin
[TRACE] 13:32:47.155 shell.go:resolveConfigPath() - 34.625µs
[TRACE] 13:32:47.155 shell.go:Init() - 430.792µs
[TRACE] 13:32:47.155 shell.go:Flags() - 42ns
[TRACE] 13:32:47.156 config.go:loadConfig() - 779.625µs
[TRACE] 13:32:47.156 shell.go:Flags() - 41ns
[DEBUG] 13:32:47.156 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.156 shell.go:Getenv(OMP_CACHE_DISABLED) - 2.542µs
[DEBUG] 13:32:47.156 shell.go:Shell:646 → no shell name provided in flags, trying to detect it
[DEBUG] 13:32:47.156 shell.go:Shell:654 → process name: zsh
[TRACE] 13:32:47.156 shell.go:Shell() - 226.792µs
[DEBUG] 13:32:47.156 shell.go:Getenv:392 → 5.9
[TRACE] 13:32:47.156 shell.go:Getenv(POSH_SHELL_VERSION) - 3.208µs
[DEBUG] 13:32:47.156 debug.go:PrintDebug:22 → Segment: Title
[DEBUG] 13:32:47.156 text.go:Render:70 → Rendering template: {{ .Shell }} in {{ .Folder }}
[TRACE] 13:32:47.156 shell_unix.go:Root() - 1.084µs
[TRACE] 13:32:47.156 shell.go:Shell() - 42ns
[TRACE] 13:32:47.156 shell.go:StatusCodes() - 333ns
[DEBUG] 13:32:47.156 shell_unix.go:IsWsl:35 → false
[TRACE] 13:32:47.156 shell_unix.go:IsWsl() - 23.5µs
[DEBUG] 13:32:47.156 shell.go:TemplateCache:827 → environment: [TERM_SESSION_ID=w0t0p0:2A1E35FC-21A9-4180-9A4A-6DDD0E29156E SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.0UqnkJboBM/Listeners LC_TERMINAL_VERSION=3.4.21 COLORFGBG=7;0 ITERM_PROFILE=MWD SQLITE_EXEMPT_PATH_FROM_VNODE_GUARDS=/Users/martin.diers/Library/WebKit/Databases XPC_FLAGS=0x0 LANG=en_US.UTF-8 PWD=/Users/martin.diers SHELL=/bin/zsh __CFBundleIdentifier=com.googlecode.iterm2 TERM_PROGRAM_VERSION=3.4.21 TERM_PROGRAM=iTerm.app PATH=/Users/martin.diers/.asdf/shims:/opt/homebrew/opt/asdf/libexec/bin:/opt/homebrew/opt/postgresql@15/bin:/Users/martin.diers/.local/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/martin.diers/.cask/bin:/Users/martin.diers/bin:/Users/martin.diers/.gem/ruby/2.0.0/bin:/Users/martin.diers/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/usr/local/munki LC_TERMINAL=iTerm2 COLORTERM=truecolor COMMAND_MODE=unix2003 TERM=xterm-256color HOME=/Users/martin.diers TMPDIR=/var/folders/yp/n8_rd_jd70nc6n7vwtr88w2c0000gp/T/ USER=martin.diers XPC_SERVICE_NAME=0 LOGNAME=martin.diers ITERM_SESSION_ID=w0t0p0:2A1E35FC-21A9-4180-9A4A-6DDD0E29156E __CF_USER_TEXT_ENCODING=0x1F6:0x0:0x0 SHLVL=1 OLDPWD=/Users/martin.diers MANPATH=/opt/homebrew/share/man: MANPAGER=/opt/homebrew/bin/most MOST_EDITOR=/opt/homebrew/bin/nvim %s HOMEBREW_GITHUB_API_TOKEN=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 HOMEBREW_INSTALL_CLEANUP=1 HOMEBREW_VERBOSE=1 LIBRARY_PATH=/opt/homebrew/lib PAGER=most EDDITOR=nvim POSH_THEME=/opt/homebrew/opt/oh-my-posh/themes/powerlevel10k_rainbow.omp.json POSH_SHELL_VERSION=5.9 POSH_PID=77552 POWERLINE_COMMAND=oh-my-posh CONDA_PROMPT_MODIFIER=false POSH_PROMPT_COUNT=5 ASDF_DIR=/opt/homebrew/opt/asdf/libexec _=/opt/homebrew/bin/oh-my-posh]
[TRACE] 13:32:47.156 shell.go:GOOS() - 291ns
[DEBUG] 13:32:47.156 shell.go:Pwd:422 → /Users/martin.diers
[TRACE] 13:32:47.156 shell.go:Pwd() - 23.667µs
[TRACE] 13:32:47.156 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 83ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 41ns
[DEBUG] 13:32:47.156 shell.go:User:557 → martin.diers
[TRACE] 13:32:47.156 shell.go:User() - 2.5µs
[DEBUG] 13:32:47.156 shell.go:Host:569 → ML20014
[TRACE] 13:32:47.156 shell.go:Host() - 9.125µs
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[DEBUG] 13:32:47.156 shell.go:Getenv:392 → 1
[TRACE] 13:32:47.156 shell.go:Getenv(SHLVL) - 2.125µs
[TRACE] 13:32:47.156 shell.go:TemplateCache() - 147.708µs
[DEBUG] 13:32:47.156 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.156 shell.go:Getenv(POSH_CURSOR_LINE) - 20.083µs
[DEBUG] 13:32:47.156 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.156 shell.go:Getenv(POSH_CURSOR_COLUMN) - 1.791µs
[TRACE] 13:32:47.156 shell.go:Flags() - 41ns
[TRACE] 13:32:47.156 shell.go:Flags() - 0s
[TRACE] 13:32:47.156 shell.go:Shell() - 42ns
[TRACE] 13:32:47.156 shell.go:Shell() - 42ns
[TRACE] 13:32:47.156 shell.go:Flags() - 84ns
[TRACE] 13:32:47.156 shell.go:Flags() - 167ns
[TRACE] 13:32:47.156 shell.go:Pwd() - 83ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[DEBUG] 13:32:47.156 segment.go:SetEnabled:497 → Segment: Git
[TRACE] 13:32:47.156 shell.go:Pwd() - 83ns
[TRACE] 13:32:47.156 shell.go:Flags() - 166ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 83ns
[DEBUG] 13:32:47.156 shell_unix.go:IsWsl:35 → false
[TRACE] 13:32:47.156 shell_unix.go:IsWsl() - 2.834µs
[DEBUG] 13:32:47.156 segment.go:SetEnabled:497 → Segment: Os
[TRACE] 13:32:47.156 shell_unix.go:IsWsl2() - 5.375µs
[TRACE] 13:32:47.156 shell.go:Pwd() - 166ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.156 shell.go:GOOS() - 42ns
[DEBUG] 13:32:47.156 segment.go:SetEnabled:497 → Segment: Path
[TRACE] 13:32:47.156 shell.go:Pwd() - 84ns
[TRACE] 13:32:47.156 shell.go:Shell() - 84ns
[TRACE] 13:32:47.156 shell.go:Shell() - 83ns
[DEBUG] 13:32:47.156 properties.go:GetString:28 → 
[TRACE] 13:32:47.156 shell.go:TemplateCache() - 125ns
[DEBUG] 13:32:47.156 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 13:32:47.157 properties.go:GetString:28 → 
[DEBUG] 13:32:47.157 properties.go:GetString:28 → 
[TRACE] 13:32:47.157 shell.go:GOOS() - 83ns
[DEBUG] 13:32:47.157 properties.go:GetString:28 → ~
[DEBUG] 13:32:47.157 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 13:32:47.157 shell.go:PathSeparator() - 84ns
[TRACE] 13:32:47.157 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.157 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.157 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.157 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.157 shell.go:PathSeparator() - 42ns
[TRACE] 13:32:47.157 shell.go:PathSeparator() - 42ns
[TRACE] 13:32:47.157 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.157 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.157 shell.go:PathSeparator() - 84ns
[TRACE] 13:32:47.157 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.157 shell.go:GOOS() - 84ns
[TRACE] 13:32:47.157 shell.go:PathSeparator() - 41ns
[DEBUG] 13:32:47.157 shell.go:CommandPath:608 → /usr/bin/git
[TRACE] 13:32:47.157 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.157 shell.go:CommandPath(git) - 146.5µs
[TRACE] 13:32:47.157 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.157 shell.go:HasCommand(git) - 149.25µs
[TRACE] 13:32:47.157 shell.go:PathSeparator() - 42ns
[TRACE] 13:32:47.157 shell.go:Pwd() - 83ns
[TRACE] 13:32:47.157 shell.go:Pwd() - 41ns
[DEBUG] 13:32:47.157 shell_unix.go:IsWsl:35 → false
[TRACE] 13:32:47.157 shell_unix.go:IsWsl() - 3.208µs
[TRACE] 13:32:47.157 shell.go:StackCount() - 42ns
[TRACE] 13:32:47.157 shell_unix.go:DirIsWritable(/Users/martin.diers) - 27.25µs
[TRACE] 13:32:47.157 shell.go:TemplateCache() - 125ns
[ERROR] 13:32:47.157 shell.go:HasParentFilePath:742 → stat .git: no such file or directory
[TRACE] 13:32:47.157 shell.go:HasParentFilePath(.git) - 54.791µs
[DEBUG] 13:32:47.157 properties.go:GetBool:22 → fetch_bare_info: false
[DEBUG] 13:32:47.157 text.go:Render:70 → Rendering template:   {{ .Path }}
[DEBUG] 13:32:47.157 text.go:Render:70 → Rendering template:  {{ if .WSL }}WSL at {{ end }}{{.Icon}}
[TRACE] 13:32:47.158 shell.go:TemplateCache() - 417ns
[TRACE] 13:32:47.158 shell.go:Shell() - 41ns
[TRACE] 13:32:47.158 shell.go:TemplateCache() - 83ns
[TRACE] 13:32:47.158 shell.go:Shell() - 42ns
[DEBUG] 13:32:47.158 text.go:Render:70 → Rendering template: diamond
[DEBUG] 13:32:47.158 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.158 text.go:Render:70 → Rendering template: powerline
[TRACE] 13:32:47.158 shell.go:Shell() - 83ns
[TRACE] 13:32:47.158 shell.go:Shell() - 42ns
[TRACE] 13:32:47.158 shell.go:Shell() - 42ns
[TRACE] 13:32:47.158 shell.go:Shell() - 42ns
[TRACE] 13:32:47.158 shell.go:Flags() - 291ns
[TRACE] 13:32:47.158 shell.go:Pwd() - 125ns
[TRACE] 13:32:47.158 shell.go:GOOS() - 42ns
[DEBUG] 13:32:47.158 segment.go:SetEnabled:497 → Segment: Time
[DEBUG] 13:32:47.158 properties.go:GetString:28 → 15:04:05
[TRACE] 13:32:47.158 shell.go:Flags() - 167ns
[TRACE] 13:32:47.158 shell.go:Pwd() - 42ns
[TRACE] 13:32:47.158 shell.go:TemplateCache() - 83ns
[TRACE] 13:32:47.158 shell.go:GOOS() - 41ns
[TRACE] 13:32:47.158 shell.go:Flags() - 84ns
[TRACE] 13:32:47.158 shell.go:Flags() - 125ns
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → extensions: [*.rb Rakefile Gemfile]
[DEBUG] 13:32:47.158 segment.go:SetEnabled:497 → Segment: Root
[TRACE] 13:32:47.158 shell.go:Pwd() - 42ns
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → folders: []
[TRACE] 13:32:47.158 shell.go:GOOS() - 41ns
[DEBUG] 13:32:47.158 segment.go:SetEnabled:497 → Segment: Executiontime
[TRACE] 13:32:47.158 shell_unix.go:Root() - 542ns
[TRACE] 13:32:47.158 shell.go:Pwd() - 125ns
[TRACE] 13:32:47.158 shell.go:GOOS() - 208ns
[DEBUG] 13:32:47.158 segment.go:SetEnabled:497 → Segment: Azfunc
[TRACE] 13:32:47.158 shell.go:Flags() - 42ns
[TRACE] 13:32:47.158 shell.go:Pwd() - 41ns
[TRACE] 13:32:47.158 shell.go:Flags() - 42ns
[DEBUG] 13:32:47.158 properties.go:GetBool:22 → always_enabled: false
[TRACE] 13:32:47.158 shell.go:ExecutionTime[TRACE] 13:32:47.158 shell.go:Pwd() - 41ns
[TRACE] 13:32:47.158 shell.go:GOOS() - 83ns
[TRACE] 13:32:47.158 shell.go:GOOS() - 42ns
[DEBUG] 13:32:47.158 segment.go:SetEnabled:497 → Segment: Go
[DEBUG] 13:32:47.158 properties.go:GetFloat64:34 → threshold: 500.000000
[TRACE] 13:32:47.158 shell.go:Flags() - 792ns
[TRACE] 13:32:47.158 shell.go:Flags() - 208ns
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → extensions: [*.jl]
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → extensions: [host.json local.settings.json function.json]
[TRACE] 13:32:47.158 shell.go:Flags() - 41ns
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → extensions: [*.go go.mod]
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:32:47.158 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → folders: []
[TRACE] 13:32:47.158 shell.go:Pwd() - 83ns
[DEBUG] 13:32:47.158 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:32:47.158 shell.go:Pwd() - 42ns
[DEBUG] 13:32:47.158 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:32:47.158 shell.go:GOOS() - 42ns
[TRACE] 13:32:47.158 shell.go:Pwd() - 42ns
[TRACE] 13:32:47.158 shell.go:Pwd() - 42ns
[DEBUG] 13:32:47.158 segment.go:SetEnabled:497 → Segment: Node
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .npmrc .vue]
[TRACE] 13:32:47.158 shell.go:Pwd() - 250ns
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:32:47.158 properties.go:GetString:28 → {{ .Code }}
[DEBUG] 13:32:47.158 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:32:47.158 shell.go:Pwd() - 292ns
[TRACE] 13:32:47.158 shell.go:Pwd() - 83ns
[DEBUG] 13:32:47.158 segment.go:SetEnabled:497 → Segment: Aws
[DEBUG] 13:32:47.158 properties.go:GetBool:22 → display_default: false
[DEBUG] 13:32:47.158 segment.go:SetEnabled:497 → Segment: Status
[DEBUG] 13:32:47.158 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.158 shell.go:StatusCodes() - 41ns
[TRACE] 13:32:47.158 shell.go:Getenv(AWS_VAULT) - 2.667µs
[DEBUG] 13:32:47.158 shell.go:Getenv:392 → NO DATA
[DEBUG] 13:32:47.158 text.go:Render:70 → Rendering template: {{ .Code }}
[TRACE] 13:32:47.158 shell.go:Getenv(AWS_DEFAULT_PROFILE) - 2µs
[DEBUG] 13:32:47.158 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.158 shell.go:Getenv(AWS_PROFILE) - 1.792µs
[DEBUG] 13:32:47.158 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.158 shell.go:Getenv(AWS_REGION) - 2.042µs
[DEBUG] 13:32:47.158 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.158 shell.go:Getenv(AWS_DEFAULT_REGION) - 2µs
[DEBUG] 13:32:47.158 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.158 shell.go:Getenv(AWS_CONFIG_FILE) - 2.125µs
[TRACE] 13:32:47.158 shell.go:Flags() - 291ns
[DEBUG] 13:32:47.158 properties.go:GetString:28 → files
[TRACE] 13:32:47.158 shell.go:Pwd() - 83ns
[TRACE] 13:32:47.158 shell.go:GOOS() - 42ns
[DEBUG] 13:32:47.158 segment.go:SetEnabled:497 → Segment: Python
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 13:32:47.158 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv env venv-win pyenv-win]
[DEBUG] 13:32:47.158 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:32:47.158 shell.go:Pwd() - 42ns
[DEBUG] 13:32:47.158 shell.go:FileContent:531 ↓
    [default]
    region = us-east-1
    output = json

    [profile legacy]
    region = us-east-1
    output = json

[TRACE] 13:32:47.158 shell.go:FileContent(/Users/martin.diers/.aws/config) - 104.958µs
[TRACE] 13:32:47.158 shell.go:TemplateCache() - 500ns
[DEBUG] 13:32:47.158 properties.go:GetBool:22 → always_enabled: true
[TRACE] 13:32:47.158 shell.go:TemplateCache() - 42ns
[DEBUG] 13:32:47.158 text.go:Render:70 → Rendering template:  {{ .CurrentDate | date .Format }} 
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template:  {{ if gt .Code 0 }}{{ reason .Code }}{{ else }}✔{{ end }}
[TRACE] 13:32:47.159 shell.go:TemplateCache() - 42ns
[TRACE] 13:32:47.159 shell.go:Shell() - 208ns
[TRACE] 13:32:47.159 shell.go:TemplateCache() - 708ns
[TRACE] 13:32:47.159 shell.go:Shell() - 42ns
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: {{ if gt .Code 0 }}#cc2222{{ end }}
[TRACE] 13:32:47.159 shell.go:TemplateCache() - 375ns
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: powerline
[DEBUG] 13:32:47.159 text.go:Render:70 → Rendering template: diamond
[DEBUG] 13:32:47.199 shell_unix.go:TerminalWidth:70 → terminal width: 149
[TRACE] 13:32:47.199 shell_unix.go:TerminalWidth() - 39.6905ms
[TRACE] 13:32:47.199 shell.go:Shell() - 42ns
[TRACE] 13:32:47.199 shell.go:Shell() - 83ns
[TRACE] 13:32:47.199 shell.go:Shell() - 42ns
[TRACE] 13:32:47.199 shell.go:Shell() - 42ns
[TRACE] 13:32:47.199 shell.go:Flags() - 83ns
[TRACE] 13:32:47.199 shell.go:Pwd() - 83ns
[TRACE] 13:32:47.199 shell.go:GOOS() - 83ns
[DEBUG] 13:32:47.199 segment.go:SetEnabled:497 → Segment: Text
[TRACE] 13:32:47.199 shell.go:TemplateCache() - 6.583µs
[DEBUG] 13:32:47.199 text.go:Render:70 → Rendering template: ╰─
[TRACE] 13:32:47.199 shell.go:Shell() - 83ns
[DEBUG] 13:32:47.199 shell.go:Getenv:392 → iTerm.app
[TRACE] 13:32:47.199 shell.go:Getenv(TERM_PROGRAM) - 26.417µs
[DEBUG] 13:32:47.199 text.go:Render:70 → Rendering template: plain
[TRACE] 13:32:47.199 shell.go:Shell() - 42ns
[TRACE] 13:32:47.199 shell.go:Shell() - 42ns
[TRACE] 13:32:47.199 shell.go:Shell() - 41ns
[TRACE] 13:32:47.199 shell.go:Shell() - 42ns
[TRACE] 13:32:47.199 shell.go:Flags() - 125ns
[TRACE] 13:32:47.199 shell.go:Pwd() - 83ns
[TRACE] 13:32:47.199 shell.go:GOOS() - 83ns
[DEBUG] 13:32:47.199 segment.go:SetEnabled:497 → Segment: Text
[TRACE] 13:32:47.199 shell.go:TemplateCache() - 166ns
[DEBUG] 13:32:47.199 text.go:Render:70 → Rendering template: ─╯
[TRACE] 13:32:47.199 shell.go:Shell() - 41ns
[DEBUG] 13:32:47.200 text.go:Render:70 → Rendering template: plain
[TRACE] 13:32:47.200 shell.go:Shell() - 41ns
[TRACE] 13:32:47.200 shell.go:Shell() - 42ns
[TRACE] 13:32:47.200 shell.go:Flags() - 42ns
[TRACE] 13:32:47.200 shell.go:Shell() - 42ns
[TRACE] 13:32:47.200 shell.go:Shell() - 42ns
[TRACE] 13:32:47.200 shell.go:Flags() - 208ns
[DEBUG] 13:32:47.200 shell.go:Getenv:392 → NO DATA
[TRACE] 13:32:47.200 shell.go:Getenv(XDG_CACHE_HOME) - 2.583µs
[TRACE] 13:32:47.200 shell_unix.go:CachePath() - 29.334µs
[TRACE] 13:32:47.200 shell.go:Flags() - 41ns
```

--- a/src/engine/block.go
+++ b/src/engine/block.go
@@ -129,14 +129,18 @@ func (b *Block) RenderSegments() (string, int) {
 		if !segment.Enabled && segment.style() != Accordion {
 			continue
 		}
+
 		if colors, newCycle := cycle.Loop(); colors != nil {
 			cycle = &newCycle
 			segment.colors = colors
 		}
+
 		b.setActiveSegment(segment)
 		b.renderActiveSegment()
 	}
+
 	b.writeSeparator(true)
+
 	return b.writer.String()
 }
 

--- a/src/engine/engine.go
+++ b/src/engine/engine.go
@@ -222,6 +222,14 @@ func (e *Engine) renderBlock(block *Block, cancelNewline bool) bool {
 			return false
 		}
 
+		// in ZSH, RPROMPT is printed with a trailing space
+		// to ensure alignment, we need to print a space here
+		// see https://github.com/JanDeDobbeleer/oh-my-posh/issues/4327
+		if e.Env.Shell() == shell.ZSH {
+			text += " "
+			length++
+		}
+
 		space, OK := e.canWriteRightBlock(false)
 		// we can't print the right block as there's not enough room available
 		if !OK {

